### ISSUE TITLE
Native crash on sendMessage

### ISSUE THREAD

**Issue by InI4** [twilio/twilio-chat-demo-android#58]
### Description

Sending a message results in  a native crash.
This happens in my branch, that I created by moving from 2.0.11 and fixing all the API changes.

#### Expected Behavior

Sending message, updating thread view.

#### Actual Behavior

App immediately restarts (no crashreport or Crashlytics reports), which makes it difficult to capture below information:

```
07-15 23:23:03.872 10125-11185/de.spieleck.app.badgers.debug E/jni_mate: #
    # Fatal error in /var/lib/jenkins/jobs/chat-sdk-android/workspace/configuration/Release/label/master/sdk/build/prebuilt/jni-mate/include/ListenerWrapper.h, line 40
    # Check failed: statusListener_ && (onSuccess_ || onSuccessBackup_)
    # Invariant violated
    #
07-15 23:23:03.981 10125-11185/de.spieleck.app.badgers.debug W/google-breakpad: ### ### ### ### ### ### ### ### ### ### ### ### ###
    Chrome build fingerprint:
    67.0.3396.87
    339608752
    ### ### ### ### ### ### ### ### ### ### ### ### ###
07-15 23:23:03.982 10125-11185/de.spieleck.app.badgers.debug A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 11185 (- 10207 - 10226), pid 10125 (p.badgers.debug)
```

#### Reproduces how Often

100%

#### Chat Android SDK

3.11.0

#### Android API

Android 8.1

#### Android Device

OnePlus6

**Comment by berkus** [comment]
As discussed in #45 this may be due to actual listener going away.

**Comment by InI4** [comment]
As discussed in #45 the main point seems to have been (tests moved forward to 4.0 rc though) that I didn't provide a CallbackHandler for an event I didn't know what it is good for what to use for and didn't know I have to provide, otherwise things crash deeply in native code. So all fine here now, except API annotations would have helped.

**Comment by berkus** [comment]
Same update as in #45: Added annotations, they will be released with the next 4.0.0 drop.
